### ISSUE TITLE
[Errno 111] Connection refused when signing a transaction

### ISSUE THREAD

**Issue by 0xYao** [aws-samples/aws-nitro-enclave-blockchain-wallet#4]
- A reproducible test case or series of steps
  - I followed all the steps outlined in the [part 2 of the blog post](https://aws.amazon.com/blogs/database/part-2-aws-nitro-enclaves-for-secure-blockchain-key-management/) to deploy the whole stack using CDK. It threw the following error when I tried to sign an Ethereum transaction via the Lambda console.

  ```bash
  {
    "errorMessage": "exception happened sending decryption request to Nitro Enclave: [Errno 111] Connection refused",
    "errorType": "Exception",
    "stackTrace": [
    "  File \"/var/task/lambda_function.py\", line 123, in lambda_handler\n    raise Exception(\"exception happened sending decryption request to Nitro Enclave: {}\".format(e))\n"
    ]
  }
  ```

  - The transaction payload I used was the same as given in the example, e.g.

```
  {
    "operation": "sign_transaction",
    "transaction_payload": {
      "value": 0.0001,
      "to": "0xa5D3241A1591061F2a4bB69CA0215F66520E67cf",
      "nonce": 0,
      "type": 2,
      "chainId": 4,
      "gas": 100000,
      "maxFeePerGas": 100000000000,
      "maxPriorityFeePerGas": 3000000000
    }
  }
```

- The version of our code being used
  - Commit `1fa0d541af348f49fc6533e05232ac64b9991f93`

- Any modifications you've made relevant to the bug
  - N/A

- Anything unusual about your environment or deployment
  - I noticed health checks failed on both of my EC2 instances, but those instances were still running and I could connect to them using a session manager.

**Comment by dpdornseifer** [comment]
Hi @0xYao ,
thanks for sharing this issue. 
Is this issue related to #5 or #6  or did it happen in an unrelated deployment? Did the EC2 or load balancer health checks fail? Which region did you deploy the stack into?  

Cheers,
David

**Comment by 0xYao** [comment]
> Hi @0xYao , thanks for sharing this issue. Is this issue related to #5 or #6 or did it happen in an unrelated deployment? Did the EC2 or load balancer health checks fail? Which region did you deploy the stack into?
> 
> Cheers, David

I had to fix both #5 and #6 first to successfully deploy. The load balancer health checks failed and I deployed the stack into `us-east-1`.

**Comment by dpdornseifer** [comment]
Hi @0xYao, the load balancer health checks would just fail if the http_server docker container is not running on the EC2 instance. Can you please confirm that the docker container is up and running. 
Just to make sure that I understand you correctly, you fixed the local build issue first (#6), executed the binary build step (`./scripts/build_kmstool_enclave_cli.sh`) and executed a cdk deploy with default settings to `us-east-1`?

**Comment by ryan-ohare-nydig** [comment]
Hi, I've seen this same issue.

* A reproducible test case or series of steps
  * Part 2 of the blog post, setting post setting up the instances in cdk. The nitro-enclaves appear to come up but are in some odd state. `describe-enclaves` returns a null set despite the enclaves are up and running.
  * Evidence:
```
sh-4.2$ sudo su
[root@ip-10-0-195-156 bin]# nitro-cli describe-enclaves
[]
[root@ip-10-0-195-156 bin]# ps -ef | grep nitro-cli
root     13721 13712  0 10:49 ?        00:00:00 /bin/nitro-cli run-enclave --cpu-count 2 --memory 3806 --eif-path /home/ec2-user/app/server/signing_server.eif --enclave-cid 16
root     13730 13721  0 10:49 ?        00:00:00 [nitro-cli] <defunct>
root     13733     1 16 10:49 ?        00:00:09 /bin/nitro-cli run-enclave --cpu-count 2 --memory 3806 --eif-path /home/ec2-user/app/server/signing_server.eif --enclave-cid 16
root     13961 13753  0 10:50 pts/0    00:00:00 grep --color=auto nitro-cli
[root@ip-10-0-195-156 bin]#
```
Additionally the health checks are failing for the target groups. This appears related to the closed issues #4  and  #5.

* The version of our code being used
  * 7f96f2f0a7d2b5598fc474558bb614bccce33894

 * Any modifications you've made relevant to the bug
   * N/A
 * Anything unusual about your environment or deployment
   * Running everything in us-east-2
